app/src/citations.py

```diff
@@ -75,6 +76,37 @@ def dereference_citations(
     return citations
 
 
+CitationNumberAndSubstring: TypeAlias = dict[int, str]
+ChunkWithCitation: TypeAlias = dict[Chunk, list[CitationNumberAndSubstring]]
```

Comment:
FYI: You don't need to `import TypeAlias` but it doesn't hurt:
```suggestion
CitationNumberAndSubstring = dict[int, str]
ChunkWithCitation = dict[Chunk, list[CitationNumberAndSubstring]]
```